src/battle_ai_main.c

```diff
@@ -4150,7 +4151,7 @@ static s16 AI_CheckViability(u8 battlerAtk, u8 battlerDef, u16 move, s16 score)
         IncreaseStatUpScore(battlerAtk, battlerDef, STAT_SPDEF, &score);
         break;
     case EFFECT_SHELL_SMASH:
-        if (AI_DATA->atkHoldEffect == HOLD_EFFECT_POWER_HERB)
+        if (AI_DATA->atkHoldEffect == HOLD_EFFECT_RESTORE_STATS)
```

Comment:
Whoops, this check should not exist, I must have mistook shell smash for a two-turn move. Maybe `HOLD_EFFECT_RESTORE_STATS` should boost score by 1 if anything, but otherwise the check can be removed.